e2e/sdk.userMintable.reveal.e2e.test.ts

```diff
@@ -0,0 +1,260 @@
+        )[0];
+        return (
+          resp.total > response.total &&
+          newContractCollection !== null &&
+          newContractCollection.metadata !== null
+        );
+      },
+      120000,
+      1000,
+      'Waiting for NFT collection to be available for an user',
+    );
+    let token0: any = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '0',
+    });
+    let token1: any = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '1',
+    });
+    expect(token0.metadata.name).toEqual('Dave Starbelly');
+    expect(token0.metadata.description).toEqual(
+      'Friendly OpenSea Creature that enjoys long swims in the ocean.',
+    );
+    expect(token0.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'unrevealed' });
+    expect(token1.metadata.name).toEqual('Dave Starbelly');
+    expect(token1.metadata.description).toEqual(
+      'Friendly OpenSea Creature that enjoys long swims in the ocean.',
+    );
+    expect(token1.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'unrevealed' });
+
+    console.log('Calling reveal');
+    // Reveal metadata
+    const txResponse = await erc721UserMintable.reveal({ baseURI: revealedMetadata });
+    await txResponse.wait();
+
+    console.log('Calling getTokenMetadata1');
+    token0 = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '0',
+    });
+    console.log('Calling getTokenMetadata2');
+    token1 = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '1',
+    });
+    expect(token0.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'unrevealed' });
+    expect(token1.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'unrevealed' });
+    console.log('Calling getTokenMetadata0 with resync');
+    token0 = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '0',
+      resyncMetadata: true,
+    });
+    expect(token0.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'revealed' });
+    console.log('Calling getTokenMetadata1 with resync false');
+    token1 = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '1',
+    });
+
+    console.log(token0);
+    console.log(token1);
+    expect(token1.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'unrevealed' });
+    token1 = await sdk.api.getTokenMetadata({
+      contractAddress: erc721UserMintable.contractAddress.toLowerCase(),
+      tokenId: '1',
+      resyncMetadata: true,
+    });
+    expect(token1.metadata.attributes[0]).toEqual({ trait_type: 'type', value: 'revealed' });
+
+    await wait(
```

Comment:
Why are we using the wait here?

Reply:
just in case the tokens metadata is not updated yet, so keep looping until that's done